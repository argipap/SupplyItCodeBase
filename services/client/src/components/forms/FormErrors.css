.validation-list {
    padding-left: 25px;
}

.validation-list > li {
    display: block;
}

li:before {
    font-family: 'Glyphicons Halflings';
    font-size: 12px;
    float: left;
    margin-top: 4px;
    margin-left: -17px;
}

.error {
    color: #FF3860;
}

.error:before {
    content: "\00D7";
    color: #FF3860;
}

.success {
    color: #23D160;
}

.success:before {
    content: "\2713";
    color: #23D160;
}