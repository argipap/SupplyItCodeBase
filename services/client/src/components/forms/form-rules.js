export const registerFormRules = [
    {
        id: 1,
        field: 'username',
        name: 'Username must be greater than 3 characters.',
        valid: false
    },
    {
        id: 2,
        field: 'email',
        name: 'Email must be greater than 5 characters.',
        valid: false
    },
    {
        id: 3,
        field: 'email',
        name: 'Email must be a valid email address.',
        valid: false
    },
    {
        id: 4,
        field: 'password',
        name: 'Password must be greater than 7 characters.',
        valid: false
    }
];

export const loginFormRules = [
    {
        id: 1,
        field: 'email',
        name: 'Email is required.',
        valid: false
    },
    {
        id: 2,
        field: 'password',
        name: 'Password is required.',
        valid: false
    }
];