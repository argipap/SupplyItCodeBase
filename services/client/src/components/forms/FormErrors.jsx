import React from 'react';

import './FormErrors.css';

const FormErrors = (props) => {
    return (
        <div>
            <ul className="validation-list">
                {
                    props.formRules.map((rule) => {
                        return <li
                            className={rule.valid ? "success" : "error"} key={rule.id}>{rule.name}
                        </li>
                    })
                }
            </ul>
            <br/>
        </div>
    )
};

export default FormErrors;