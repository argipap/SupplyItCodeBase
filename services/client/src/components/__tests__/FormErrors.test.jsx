import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import FormErrors from '../forms/FormErrors';
import { registerFormRules, loginFormRules } from '../forms/form-rules.js';

const registerFormProps = {
    formType: 'Register',
    formRules: registerFormRules,
};

const loginFormProps = {
    formType: 'Login',
    formRules: loginFormRules,
};

test('FormErrors (with register form) renders properly', () => {
    const wrapper = shallow(<FormErrors {...registerFormProps} />);
    const ul = wrapper.find('ul');
    expect(ul.length).toBe(1);
    const li = wrapper.find('li');
    expect(li.length).toBe(4);
    expect(li.get(0).props.children).toContain(
        'Username must be greater than 3 characters.');
    expect(li.get(1).props.children).toContain(
        'Email must be greater than 5 characters.');
    expect(li.get(2).props.children).toContain(
        'Email must be a valid email address.');
    expect(li.get(3).props.children).toContain(
        'Password must be greater than 7 characters.');
});

test('FormErrors (with register form) renders a snapshot properly', () => {
    const tree = renderer.create(
        <Router><FormErrors {...registerFormProps} /></Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

test('FormErrors (with login form) renders properly', () => {
    const wrapper = shallow(<FormErrors {...loginFormProps} />);
    const ul = wrapper.find('ul');
    expect(ul.length).toBe(1);
    const li = wrapper.find('li');
    expect(li.length).toBe(2);
    expect(li.get(0).props.children).toContain(
        'Email is required.');
    expect(li.get(1).props.children).toContain(
        'Password is required.');
});

test('FormErrors (with login form) renders a snapshot properly', () => {
    const tree = renderer.create(
        <Router><FormErrors {...loginFormProps} /></Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});