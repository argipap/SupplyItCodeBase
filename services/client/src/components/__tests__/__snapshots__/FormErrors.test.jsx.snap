// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormErrors (with login form) renders a snapshot properly 1`] = `
<div>
  <ul
    className="validation-list"
  >
    <li
      className="error"
    >
      Email is required.
    </li>
    <li
      className="error"
    >
      Password is required.
    </li>
  </ul>
  <br />
</div>
`;

exports[`FormErrors (with register form) renders a snapshot properly 1`] = `
<div>
  <ul
    className="validation-list"
  >
    <li
      className="error"
    >
      Username must be greater than 3 characters.
    </li>
    <li
      className="error"
    >
      Email must be greater than 5 characters.
    </li>
    <li
      className="error"
    >
      Email must be a valid email address.
    </li>
    <li
      className="error"
    >
      Password must be greater than 7 characters.
    </li>
  </ul>
  <br />
</div>
`;
